test/distributed/org/apache/cassandra/distributed/upgrade/MixedModeColumnMaskingTest.java

```diff
@@ -0,0 +1,113 @@
+import org.apache.cassandra.distributed.api.ICoordinator;
+import org.assertj.core.api.Assertions;
+
+import static org.apache.cassandra.distributed.api.ConsistencyLevel.ALL;
+import static org.apache.cassandra.distributed.api.Feature.GOSSIP;
+import static org.apache.cassandra.distributed.shared.AssertUtils.assertRows;
+import static org.apache.cassandra.distributed.shared.AssertUtils.row;
+
+/**
+ * Tests that dynamic data masking (DDM) functions can be attached to table columns during a rolling upgrade involving
+ * nodes that don't include DDM.
+ */
+public class MixedModeColumnMaskingTest extends UpgradeTestBase
+{
+    @Test
+    public void testColumnMasking() throws Throwable
+    {
+        new TestCase()
+        .nodes(2)
+        .nodesToUpgrade(1, 2)
+        .upgradesToCurrentFrom(v30)
+        .withConfig(config -> config.with(GOSSIP))
+        .setup(cluster -> {
+            cluster.schemaChange(withKeyspace("CREATE TABLE %s.t (k int PRIMARY KEY, v int)"));
+            cluster.coordinator(1).execute(withKeyspace("INSERT INTO %s.t(k, v) VALUES (0, 7)"), ALL);
+        })
+        .runAfterNodeUpgrade((cluster, node) -> {
+            if (node == 1)
+            {
+                ICoordinator coordinator = cluster.coordinator(1);
+
+                // create table with masked column
+                assertFails(coordinator,
+                            "CREATE TABLE %s.t1 (k int PRIMARY KEY, v int MASKED WITH DEFAULT)",
+                            "Cannot create a table with data masking functions during rolling upgrade");
+
+                // mask existing column
+                assertFails(coordinator,
+                            "ALTER TABLE %s.t ALTER v MASKED WITH DEFAULT",
+                            "Cannot add masking function to column during rolling upgrade");
+                assertColumnValue(cluster, "t", "v", 7);
+
+                // unmask existing column
+                assertFails(coordinator,
+                            "ALTER TABLE %s.t ALTER v DROP MASKED",
+                            "Cannot remove masking function from column during rolling upgrade");
+                assertColumnValue(cluster, "t", "v", 7);
+
+                // add new masked column
+                assertFails(coordinator,
+                            "ALTER TABLE %s.t ADD v2 int MASKED WITH DEFAULT",
+                            "Cannot add column with masking function during rolling upgrade");
+                assertColumnValue(cluster, "t", "v", 7);
+            }
+        }).runAfterClusterUpgrade(cluster -> {
+
+            // create table with masked column
+            cluster.schemaChange(withKeyspace("CREATE TABLE %s.t2 (k int PRIMARY KEY, v int MASKED WITH DEFAULT)"));
+            cluster.coordinator(1).execute(withKeyspace("INSERT INTO %s.t2(k, v) VALUES (0, 7)"), ALL);
+            assertColumnValue(cluster, "t2", "v", 0);
+
+            // mask existing column
```

Comment:
Comment is wrong?

Reply:
It is. Or, more exactly, the block of code immediately below it is swapped with the next block of code :) 

The intention is first masking an unmasked column, then unmasking it. Fixed.